plugin/src/main/java/org/opensearch/sql/plugin/catalog/CatalogSettings.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.plugin.catalog;
+
+import java.io.InputStream;
+import org.opensearch.common.settings.SecureSetting;
+import org.opensearch.common.settings.Setting;
+
+public class CatalogSettings {
+
+  public static final Setting<Boolean> FEDERATION_ENABLED = Setting.boolSetting(
```

Comment:
I'm wondering the use case of this setting.

Reply:
Even my thoughts were to remove this setting eventually. Now, I don't see a point having it even in the initial release. Earlier I had thoughts to use this while parsing the table Source.

Will remove if there is no impact.